<launch>

    <arg name="h_lower" default="85"/>  
    <arg name="s_lower" default="44"/>  
    <arg name="v_lower" default="126"/> 

    <arg name="h_upper" default="151"/>  
    <arg name="s_upper" default="114"/>  
    <arg name="v_upper" default="255"/>
    <arg name="v1"      default="0.56"/>  
    <arg name="v2"      default="0.46"/>  
    <arg name="v3"      default="0.28"/>
    <arg name="v4"      default="0.20"/>
    <arg name="test_mode"      default="False"/>
    <!-- node work in test mode or normal mode -->

    <node pkg="robot_vision" name="linefollow" type="line_detector.py" output="screen">
        <param name="h_lower"            value="$(arg h_lower)"/>
        <param name="s_lower"            value="$(arg s_lower)"/>
        <param name="v_lower"            value="$(arg v_lower)"/>
        <param name="h_upper"            value="$(arg h_upper)"/>
        <param name="s_upper"            value="$(arg s_upper)"/>
        <param name="v_upper"            value="$(arg v_upper)"/>
        <param name="v1"                 value="$(arg v1)"/>
        <param name="v2"                 value="$(arg v2)"/>
        <param name="v3"                 value="$(arg v3)"/>
        <param name="v4"                 value="$(arg v4)"/>
        <param name="test_mode"          value="$(arg test_mode)"/>
    </node>
    
    <node name="republish_mask" pkg="image_transport" type="republish" args="raw in:=mask_image compressed out:=mask_image" />
    <node name="republish_result" pkg="image_transport" type="republish" args="raw in:=result_image compressed out:=result_image" />
    

</launch>
